e2e/app/element/base-element.ts

```diff
@@ -108,33 +111,22 @@ export default class BaseElement {
 
   /**
    * <pre>
-   *  Check if the element is visible
+   *  Check both boxModel and style for visibility to determine whether element is visible.
    * </pre>
-   * @param {Page} page
-   * @param {ElementHandle} element
    */
   async isVisible(): Promise<boolean> {
-    return this.asElementHandle()
-      .then((elemt) => {
-        return elemt.boxModel();
-      })
-      .then((box) => {
-        return box !== null;
-      });
-  }
-
-  /**
-   * Check both boxModel and style for visibility.
-   */
-  async isDisplayed(): Promise<boolean> {
-    const elemt = await this.asElementHandle();
-    const isVisibleHandle = await this.page.evaluateHandle((e) => {
-      const style = window.getComputedStyle(e);
-      return style && style.display !== 'none' && style.visibility !== 'hidden' && style.opacity !== '0';
-    }, elemt);
-    const jValue = await isVisibleHandle.jsonValue();
-    const boxModelValue = await elemt.boxModel();
-    return jValue && boxModelValue !== null;
+    try {
+      const element = await this.asElementHandle();
+      const isVisibleHandle = await this.page.evaluateHandle((e) => {
+        const style = window.getComputedStyle(e);
+        return style && style.display !== 'none' && style.visibility !== 'hidden' && style.opacity !== '0';
+      }, element);
+      const visible = await isVisibleHandle.jsonValue();
+      const box = await element.boxModel();
+      return !!(visible && box);
+    } catch (err) {
+      return false;
```

Comment:
a blanket catch over the whole function like this could suppress programming errors. Are there specific errors we're expecting here and are there any other ways of handling them? Alternatively, can the scope of the try/catch be reduced to a specific area of concern?